### PR TITLE
Add initial node getting started doc

### PR DESCRIPTION
**Pull request by beeme1mr**
## This PR

- adds a Getting Started walk through for Node.JS

Signed-off-by: Michael Beemer <beeme1mr@users.noreply.github.com>

### CODE REVIEW

File: docs/getting-started/node.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+});
+
+app.listen(port, () => {
+  console.log(`⚡️[server]: Server is running at http://localhost:${port}`);
+});
+```
+
+### Step 3: Add the OpenFeature SDK
+
+Using NPM, install the OpenFeature SDK using the following command.
```
Comment (Kavindu-Dodan) [inline docs/getting-started/node.md:55]: Mino - we could improve the tone a little bit.

ex: - Let’s install the OpenFeature SDK with the following command,

File: docs/getting-started/node.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+npm install @openfeature/js-sdk
+```
+
+Update `index.ts` to import the SDK.
+
+```diff
+import express from "express";
++ import { OpenFeature } from "@openfeature/js-sdk";
+
+const app = express();
```
Comment (Kavindu-Dodan) [inline docs/getting-started/node.md:67]: we could remove this line and only show the import. IMO this makes the change clear

File: docs/getting-started/node.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+npm install @openfeature/flagd-provider
+```
+
+Now, update `index.ts` to import the `flagd provider`.
+
+```diff
+import { OpenFeature } from "@openfeature/js-sdk";
++ import { FlagdProvider } from "@openfeature/flagd-provider";
+
+const app: Express = express();
```
Comment (Kavindu-Dodan) [inline docs/getting-started/node.md:132]: we could remove this line and only show the import. IMO this makes the change clear

File: docs/getting-started/node.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+### Step 4: Run the application
+
+Let's start the app and see it in action. Run the following command to start the server.
+
+```sh
+npx ts-node index.ts
+```
+
+Open your favorite browser and navigate to http://localhost:8080. If all goes as planned, you should see "Express + TypeScript Server" in glorious monochrome.
```
Comment (Kavindu-Dodan) [inline docs/getting-started/node.md:106]: We could add URLs as links ex : [http://localhost:8080](http://localhost:8080)

**Comment by beeme1mr** [comment]
Here's a direct link to the doc:

https://deploy-preview-182--lucky-creponne-baf9b3.netlify.app/docs/getting-started/node/

**Review by toddbaert — APPROVED** [review]
This all looks good to me, nice job.

**Comment by moredip** [comment]
@beeme1mr I’m late in getting my feedback in, but I was wondering how you’d feel about removing the typescript-specific stuff from the tutorial. The concern is that if someone doesn’t know anything about TS they might read this and think that they have to use TS to use the JS SDK. Given that the code samples aren’t really using types anyway, it seems like it would just remove a bit of extra cognitive load if the tutorial skipped references to things like `tsc`, `ts-node`, `@types/express`, and so on.

**Comment by toddbaert** [comment]
> @beeme1mr I’m late in getting my feedback in, but I was wondering how you’d feel about removing the typescript-specific stuff from the tutorial. The concern is that if someone doesn’t know anything about TS they might read this and think that they have to use TS to use the JS SDK. Given that the code samples aren’t really using types anyway, it seems like it would just remove a bit of extra cognitive load if the tutorial skipped references to things like `tsc`, `ts-node`, `@types/express`, and so on.

@moredip 

I see you point, and also as somebody who wrote almost exclusively JS/TS for a chunk of time, knowing TS has first-class support immediately makes a project more attractive.

I'm working on extracting some of the common language/structure between this and the Java tutorial anyway. When I'm doing that I could also add a pure-JS version. What do you think of that? Is that overkill? Should we just white the JS version and mention we support TS?

**Comment by beeme1mr** [comment]
I think it might be worth having both versions. They could live in the same doc and you could toggle between the two using a tab group. I could add that next week.

**Comment by toddbaert** [comment]
> I think it might be worth having both versions. They could live in the same doc and you could toggle between the two using a tab group. I could add that next week.

I like that solution.

**Pull request merged**